collect_executables.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# Usage: ./collect_executables.sh [channel]
+# Collects geckodriver and Fx, default channel is Beta.
+
+GECKO_LINK=$(curl https://api.github.com/repos/mozilla/geckodriver/releases/latest | jq '.["assets"][0]["browser_download_url"]' | tr -d '"')
+
+UNAME_A=$(uname -a)
+if [ -n "$WSL_DISTRO_NAME" ]
+then
+    SYSTEM_NAME="win"
+else
+    if [[ $UNAME_A == *"Darwin"* ]]
+    then
+        SYSTEM_NAME="macos"
+    else
+        SYSTEM_NAME="linux"
+    fi
+fi
+
+if [[ $UNAME_A == *"arm64"* ]]
+then
+    ARCH="-aarch64"
+else
+    if [[ $SYSTEM_NAME == "linux" ]] && [[ $UNAME_A == *"i386"* ]]
+    then
+        BITS="32"
+    else
+        BITS="64"
+    fi
+fi
+
+if [[ "$SYSTEM_NAME" == "win" ]] && [[ -z $ARCH ]] && [[ $BITS = "64" ]]
+then
+    echo "Intel Win64"
+    exit 2
```

Comment:
Q: why the `exit`?

Reply:
This looked like a papercut to you, but it revealed a whole iceberg of poor coding decisions on my side. Excellent find, I'm going to have to work on this.

Reply:
i wasn't sure if it was leftover debugging code or just really bad error messaging on exit; but I'm happy to have ruined your afternoon with my rogue icebergs. 🪟🥂🧊

Reply:
It's fixed now!